plan/expression_rewriter.go

```diff
@@ -764,7 +765,11 @@ func (er *expressionRewriter) rewriteVariable(v *ast.VariableExpr) {
 		er.err = errors.Trace(err)
 		return
 	}
-	er.ctxStack = append(er.ctxStack, datumToConstant(types.NewStringDatum(val), mysql.TypeString))
+	e := datumToConstant(types.NewStringDatum(val), mysql.TypeString)
+	e.RetType.Tp = mysql.TypeVarString
+	e.RetType.Charset = er.ctx.GetSessionVars().Systems["character_set_connection"]
```

Comment:
can "character_set_connection" be defined as a constant?

Reply:
ok